lib/rule-tester/rule-tester.js

```diff
@@ -375,6 +370,304 @@ const metaSchemaDescription = `
 \thttps://eslint.org/docs/latest/extend/custom-rules#options-schemas
 `;
 
+/*
+ * Ignored test case properties when checking for test case duplicates.
+ */
+const duplicationIgnoredParameters = new Set(["name", "errors", "output"]);
+
+/**
+ * Asserts that a value is a string or undefined.
+ * If the value is not a string or undefined, it throws an assertion error
+ * @param {any} value The value to check.
+ * @param {string?} message The error message to display if the value is not a string or undefined.
+ * @returns {void}
+ */
+function assertStringOrUndefined(
+	value,
+	message = "Value must be a string or undefined",
+) {
+	assert.ok(
+		typeof value === "string" || typeof value === "undefined",
+		message,
+	);
+}
+
+/**
+ * Asserts that a value is an object.
+ * If the value is not an object, it throws an assertion error.
+ * @param {any} value The value to check.
+ * @param {string?} message The error message to display if the value is not an object.
+ * @returns {void}
+ */
+function assertObject(value, message = "Value must be an object") {
+	assert.ok(value && typeof value === "object", message);
+}
+
+/**
+ * Asserts that a value is a string.
+ * If the value is not a string, it throws an assertion error.
+ * @param {any} value The value to check.
+ * @param {string?} message The error message to display if the value is not a string.
+ * @returns {void}
+ */
+function assertString(value, message = "Value must be a string") {
```

Comment:
This function would be unnecessary with the refactoring mentioned below as it would be only called once.